src/kOS.Safe/Compilation/KS/Compiler.cs

```diff
@@ -2487,8 +2497,9 @@ private void VisitOnStatement(ParseNode node)
 
             if (triggerObject.IsInitialized())
             {
-                AddOpcode(new OpcodePush(triggerObject.VariableNameOldValue));
-                AddOpcode(new OpcodePush(triggerObject.VariableName));
+                // Store the current value into the old value to prep for the first use of the ON trigger:
+                AddOpcode(new OpcodePush(triggerObject.OldValueIdentifier));
+                VisitNode(node.Nodes[1]); // the expression in the on statement.
```

Comment:
Arrgggggggg!!! dangit.  There should be an ``OpcodeEval`` inserted here (where this comment is, just after the ``VisitNode()`` call).  I didn't catch it until re-reading my own code.  The problem being that if ``node.Nodes[1]`` is nothing more than a single identifier, what goes on the stack will be something like ``$foo`` rather than the contents of ``$foo``.  For this check, we need the contents of it, just to be safe and make sure things still work if ``foo`` stops existing.

It's a super rare case where it could matter, but it could.  because triggers need the values to be global it doesn't seem likely that ``foo`` would stop existing, but it could.